### ISSUE TITLE
[java] UnitTestShouldIncludeAssert false positive when TestNG @Test.expectedException present

### ISSUE THREAD

**Issue by numeralnathan** [pmd/pmd#6188]
**Affects PMD Version:** 7.5.0

**Rule:** [UnitTestShouldIncludeAssert](https://pmd.github.io/pmd/pmd_rules_java_bestpractices.html#unittestshouldincludeassert)

**Description:** UnitTestShouldIncludeAssert flags a method without an assert but needs to recognize TestNG's `@Test.expectedExceptions` as the "assert".

**Code Sample demonstrating the issue:**

```java
import org.testng.annotations.Test;

@Test(expectedExceptions = IOException.class)             // No assert needed since this checks the functionality.
public static void alwaysThrows() throws IOException
{
   throw new IOException("expected");
}
```

**Expected outcome:** PMD reports a violation on `alwaysThrows`, but that's wrong.  That's a false positive.

**Running PMD through:** Eclipse
